src/gdbstub_impl/ext/single_register_access.rs

```diff
@@ -13,17 +13,27 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
     ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
         let handler_status = match command {
             SingleRegisterAccess::p(p) => {
-                let mut dst = [0u8; 32]; // enough for 256-bit registers
                 let reg = <T::Arch as Arch>::RegId::from_raw_id(p.reg_id);
                 let (reg_id, reg_size) = match reg {
                     // empty packet indicates unrecognized query
                     None => return Ok(HandlerStatus::Handled),
                     Some(v) => v,
                 };
-                let dst = &mut dst[0..reg_size];
-                ops.read_register(id, reg_id, dst).handle_error()?;
+                let mut buf = p.buf;
+                if let Some(size) = reg_size {
+                    buf = &mut buf[0..size.get()];
```

Comment:
This needs to be checked buffer access, or else this will introduce a panic.

```rus
buf = buf.get_mut(..size.get()).ok_or(Error::PacketBufferOverflow)?;
```

ditto for the buffer access on line 34.